lib/kitchen/directions/bake_index/v1.rb

```diff
@@ -118,10 +118,21 @@ def bake(book:)
       @metadata_elements = book.metadata.children_to_keep.copy
       @index = Index.new
 
-      book.pages.terms.each do |term_element|
+      chapters = book.chapters
+      book.pages_and_composite_pages.terms.each do |term_element|
         # Markup the term
         page = term_element.ancestor(:page)
-        term_element.id = "auto_#{page.id}_term#{term_element.count_in(:book)}"
+        term_element.id = ''
+        page_title = ''
+        if page[:'data-type'] == 'composite-page'
+          term_element.id = "auto_composite_page_term#{term_element.count_in(:book)}"
+          chapter_number = chapters.search("[id='#{term_element.id}']")&.first&.ancestor(:chapter)&.count_in(:book)
```

Comment:
Terms are not always in a chapter; they seem to be able to come from the Preface and Introduction as well. But if the term is in a composite page, then it is in a chapter. So I didn't use `book.chapters` here because that would exclude those pages. Believe me, I wouldn't have gotten the chapter number this way otherwise 🙃 

Since `chapters` is an `ElementEnumerator`, not an `Element`, `first` actually calls the base enumerator `first` method. I don't think that should be overwritten, so I think `search(...).first` has to be used here (again, 🙃)

Reply:
ah you're right on `first`